sdk/servicebus/azure_messaging_servicebus/tests/README.md

```diff
@@ -0,0 +1,229 @@
+# Azure Service Bus Live Tests
+
+This directory contains live integration tests for the Azure Service Bus SDK for Rust. These tests validate the SDK functionality against a real Azure Service Bus namespace.
+
+## Prerequisites
+
+1. **Azure Service Bus Namespace**: You need an active Azure Service Bus namespace with at least one queue configured.
+
+2. **Authentication**: You can use either connection string authentication or TokenCredential-based authentication.
+
+3. **Test Resources**: The tests require specific Azure Service Bus entities to be created beforehand.
+
+## Required Azure Resources
+
+### For Connection String Tests (`servicebus_live_tests.rs`)
+
+Create the following resources in your Service Bus namespace:
+
+- **Queue**: A standard queue for basic send/receive operations
+- **Topic**: A topic for publish/subscribe scenarios (optional)
+- **Subscription**: A subscription on the topic (optional, only needed for topic tests)
+
+### For TokenCredential Tests (`servicebus_token_credential_tests.rs`)
+
+- **Queue**: Same queue as above
+- **Azure AD App Registration** (for ClientSecretCredential tests): Optional, but recommended for comprehensive testing
+
+## Environment Variables
+
+### Connection String Authentication
+
+Set these environment variables for connection string-based tests:
+
+```bash
+# Required for all connection string tests
+export SERVICEBUS_CONNECTION_STRING="Endpoint=sb://your-namespace.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=your-key"
+export SERVICEBUS_QUEUE_NAME="test-queue"
+
+# Optional for topic/subscription tests
+export SERVICEBUS_TOPIC_NAME="test-topic"
+export SERVICEBUS_SUBSCRIPTION_NAME="test-subscription"
+```
+
+### TokenCredential Authentication
+
+Set these environment variables for TokenCredential-based tests:
+
+```bash
+# Required for all TokenCredential tests
+export SERVICEBUS_NAMESPACE="your-namespace.servicebus.windows.net"
+export SERVICEBUS_QUEUE_NAME="test-queue"
+
+# For DeveloperToolsCredential (one of these approaches):
+
+# Option 1: Service Principal with Client Secret
+export AZURE_TENANT_ID="your-tenant-id"
+export AZURE_CLIENT_ID="your-client-id"
+export AZURE_CLIENT_SECRET="your-client-secret"
+
+# Option 2: Use Azure CLI (run 'az login' first)
+# No additional environment variables needed
+
+# Option 3: Use Managed Identity (when running on Azure)
+# No additional environment variables needed
+```
+
+## Required Permissions
+
+For TokenCredential authentication, ensure your identity has the following RBAC roles:
+
+- **Azure Service Bus Data Owner** or **Azure Service Bus Data Sender** and **Azure Service Bus Data Receiver**
+- Scope: Service Bus namespace or specific queue/topic
+
+## Running the Tests
+
+### All Tests (Unit + Live)
+
+```bash
+cargo test
```

Comment:
Nit: All these commands will run over the whole workspace depending on the current directory. Might want to note at least that they should be only under the crate directory, or pass `-p azure_messaging_servicebus` just to be sure.

Reply:
Even the ones where you specify a test name will still compile all crates but then filter all tests for a simple substring match against the values you passed in. So it's still slower if outside this crate directory. Even just a note somewhere above to say something like, "The following commands assume you're running them within the `azure_messaging_servicebus` crate directory." or something like that might be good.